### PR TITLE
feat: update documentation with new provider config

### PR DESCRIPTION
**Pull request by HomelessDinosaur**
Closes: https://github.com/nitrictech/docs/issues/383

### CODE REVIEW

File: docs/reference/cli/stacks.md [hunk 1]
```diff
@@ -28,11 +28,28 @@ Follow the prompts to create a stack for your provider.
 The stack definition will be created in the root of your project as a YAML file prefixed with `nitric-` e.g. `nitric-aws-stack.yaml`
 
 ```yaml
-name: aws-stack
-provider: aws
+name: my-aws-stack
+provider: nitric/aws@0.22.1
```
Comment (tjholm) [inline docs/reference/cli/stacks.md:32]: bump this to latest prod version

File: docs/deployment.md [hunk 2]
```diff
@@ -35,23 +35,42 @@ For nitric to maintain the state of your deployment between runs you will also n
...
+
+You can point config to all functions, or to a specific function. Either by using the `default` key or the function type i.e. `memory-optimized`. The following configurations can be made:
 
 - `memory`: The memory of the compute instance in MB. Defaults to 512.
 - `timeout`: The max running time of the function in seconds. Defaults to 15.
 - `telemetry`: The percentage of requests to be sampled for tracing. Defaults to 0%.
+
+For AWS, [provisioned concurrency](https://docs.aws.amazon.com/lambda/latest/dg/provisioned-concurrency.html) can also be added:
+
+- `provisioned-concurrency`: The amount of lambdas that should remain provisioned for immediate invocation. Defaults to 0.
```
Comment (jyecusch) [inline docs/deployment.md:76]:
```suggestion
- `provisioned-concurrency`: The number of lambda instances that should remain warm for immediate invocation. Defaults to 0.
```

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: docs/deployment.md [hunk 3]
```diff
@@ -35,23 +35,42 @@ For nitric to maintain the state of your deployment between runs you will also n
...
+region: us-east-1
 config:
   default:
-    memory: 1024
-    telemetry: 10
-  functions/hello.ts:
-    timeout: 30
+    lambda:
+      memory: 1024
+      telemetry: 10
```
Comment (tjholm) [inline docs/deployment.md:49]: telemetry is top level for the category

File: docs/deployment.md [hunk 4]
```diff
@@ -35,23 +35,42 @@ For nitric to maintain the state of your deployment between runs you will also n
...
+    target: lambda
```
Comment (tjholm) [inline docs/deployment.md:50]: target is no longer used I think (need to test this to make sure)

**Review by davemooreuws — APPROVED** [review]

**Review by tjholm — APPROVED** [review]

**Pull request merged**